omod/src/main/java/org/openmrs/module/owa/activator/OwaActivator.java

```diff
@@ -163,4 +170,37 @@ public void stopped() {
 		log.info("OWA Module stopped");
 	}
 	
+	/**
+	 * Used to set the login Url Global Property that directs to the Add-on OWA UI if no UI module
+	 * is installed. Searches against the legacy and Reference Application module on installed UI
+	 * modules.
+	 */
+	private void configureLoginUrl(AppManager appManager) {
+		if (appManager.exists(ADD_ON_MANAGER_APP)) {
+			if (!uiModuleInstalled()) {
+				String launchUrl = null;
+				for (App app : appManager.getApps()) {
+					if (app.getName().equalsIgnoreCase(ADD_ON_MANAGER_APP)) {
+						launchUrl = app.getLaunchUrl();
+						if (launchUrl == null) {
+							launchUrl = "owa/" + app.getFolderName() + "/index.html";
```

Comment:
The nesting in this function is too much. What do you think of return early and often? https://jamesmonger.com/2019/08/06/return-early-return-often.html?utm_medium=email&utm_source=topic+optin&utm_campaign=awareness&utm_content=20190824+prog+nl&mkt_tok=eyJpIjoiWmpnNE5HWTFPRGhrT0dGbSIsInQiOiJRc3Q0eEhBam1IY09HMzB3Mk00aUtiNUQxQlJ0RzBvVGo1dE1QZEh4TnR0ajR6WmFGUHg3d1FpSG54cWRWck5Ca0ZPRnRVRDRUTnJUTnYwclhHQzlLZGRURjNEN1wvSW5pTDIxdEU3c1BiY0E1Z0QzaVdGeW9aZ1wvaCt0N1dOXC9IcCJ9